packages/ai-editor/src/browser/ask-ai-input-widget.ts

```diff
@@ -0,0 +1,134 @@
+// *****************************************************************************
+// Copyright (C) 2025 EclipseSource GmbH.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { Coordinate } from '@theia/core/lib/browser/context-menu-renderer';
+import { Disposable, DisposableCollection } from '@theia/core/lib/common/disposable';
+import { Emitter, Event } from '@theia/core/lib/common/event';
+
+export interface InputSubmitEvent {
+    userInput: string;
+}
+
+export class AskAIInputWidget implements Disposable {
+    private readonly element: HTMLDivElement;
+    private readonly inputElement: HTMLInputElement;
+    private readonly disposables = new DisposableCollection();
+
+    private readonly onSubmitEmitter = new Emitter<InputSubmitEvent>();
+    private readonly onCancelEmitter = new Emitter<void>();
+
+    readonly onSubmit: Event<InputSubmitEvent> = this.onSubmitEmitter.event;
+    readonly onCancel: Event<void> = this.onCancelEmitter.event;
+
+    constructor() {
+        // Create container element
+        this.element = document.createElement('div');
```

Comment:
Should this be refactored to React? I don't think Theia typically uses low-level JS Apis to be widgets